tests/integration/lifecycle/test_simple.py

```diff
@@ -40,6 +40,25 @@
 def test_actions_simple(new_dir, mocker):
     parts = yaml.safe_load(parts_yaml)
 
+    # first run
+    # command: pull
     lf = craft_parts.LifecycleManager(parts, application_name="test_demo")
     actions = lf.plan(Step.PULL)
-    assert actions == []
+    assert actions == [
+        Action("foo", Step.PULL),
+        Action("bar", Step.PULL),
+        Action("foobar", Step.PULL),
+    ]
+
+    # foobar part depends on nothing
+    # command: prime foobar
+    lf = craft_parts.LifecycleManager(parts, application_name="test_demo")
+    actions = lf.plan(Step.PRIME, ["foobar"])
+    assert actions == [
+        # FIXME: this will be skipped after the executor writes state to disk
+        # Action("foobar", Step.PULL, action_type=ActionType.SKIP, reason="already ran"),
+        Action("foobar", Step.PULL, action_type=ActionType.RUN, reason=None),
```

Comment:
Do you specify action_type and reason just to prepare for the state being written?  Or is there something else special about this?

Reply:
The only special thing is that this line will disappear and will be replaced by the line immediately above that's currently commented out. I just copied the line and replaced the values, it makes the line stand out and is an extra reminder that it should be removed later :)

Reply:
figured, thanks!